### PR TITLE
MaybeUninit: remove deprecated functions

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: src/libcore/mem.rs [hunk 1]
```diff
@@ -1152,15 +1152,6 @@ impl<T> MaybeUninit<T> {
         MaybeUninit { uninit: () }
     }
 
-    /// Deprecated before stabilization.
-    #[unstable(feature = "maybe_uninit", issue = "53491")]
-    #[inline(always)]
-    // FIXME: still used by stdsimd
```
Comment (Centril) [inline src/libcore/mem.rs]: Was this done?

**Comment by Centril** [comment]
r? @Centril 

r=me assuming it (^) was fixed and green travis

File: src/libcore/mem.rs (hunk 1 above)
Comment (RalfJung) [reply src/libcore/mem.rs]: Yes, this landed with https://github.com/rust-lang/rust/pull/59556

**Pull request merged**